tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,311 @@
+            echo "ExecStart=/bin/bash /opt/testcloud-guest.sh" >> /etc/systemd/system/testcloud.service && \
+            echo "[Install]" >> /etc/systemd/system/testcloud.service && \
+            echo "WantedBy=multi-user.target" >> /etc/systemd/system/testcloud.service && \
+            systemctl enable testcloud.service
+        '''
+
+        containerfile_parsed = render_template(
+            containerfile_template,
+            base_image=base_image,
+            testcloud_guest=simple_http_start_guest)
+        (self.workdir / 'Containerfile').write_text(containerfile_parsed)
+
+        image_tag = f'localhost/tmtmodified-{self._get_id()}'
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger, env=PODMAN_ENV if self._rootless else None)
+
+        return image_tag
+
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        """ Build the "base" or user supplied container image """
+        image_tag = f'localhost/tmtbase-{self._get_id()}'
+        self._logger.debug("Building container image")
+        tmt.utils.Command(
+            "podman", "build", self._expand_path(workdir),
+            "-f", self._expand_path(containerfile),
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger, env=PODMAN_ENV if self._rootless else None)
+        return image_tag
+
+    def _build_bootc_disk(self, containerimage: str, image_builder: str) -> None:
+        """ Build the bootc disk from a container image using bootc image builder """
+        self._logger.debug("Building bootc disk image")
+
+        tmt.utils.Command(
+            "podman", "run", "--rm", "--privileged",
+            "-v", f'{CONTAINER_STORAGE_DIR}:{CONTAINER_STORAGE_DIR}',
+            "--security-opt", "label=type:unconfined_t",
+            "-v", f"{self.workdir}:/output",
+            image_builder, "build",
+            "--type", "qcow2",
+            "--local", containerimage
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger, env=PODMAN_ENV if self._rootless else None)
+
+    def _init_podman_machine(self) -> None:
+        try:
+            tmt.utils.Command(
+                "podman", "machine", "rm", "-f", PODMAN_MACHINE_NAME
+                ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+        except:
+            self._logger.debug("Unable to remove existing podman machine (it might not exist)")
+
+        self._logger.debug("Initializing podman machine")
+        tmt.utils.Command(
+            "podman", "machine", "init", "--rootful",
+            "--disk-size", PODMAN_MACHINE_DISK_SIZE,
+            "--memory", PODMAN_MACHINE_MEM,
+            "--cpus", PODMAN_MACHINE_CPU,
+            "-v", "/var/tmp/tmt:/var/tmp/tmt",
```

Comment:
I'd make this path module-level member, some `DEFAULT_TMP_PATH` or something - the issue is, it's in the middle of code, it's not configurable here, it looks like a workdir root users **can** and often **do** override, so it will need to be configurable in the future, and linters complain it's an unsafe use of a temporary directory. It's not, it's perfectly fine, but moving it to module level, we could waive silence linter more easily.